app/models/piece.rb

```diff
@@ -15,7 +16,6 @@ def move_piece!(new_x, new_y)
     transaction do
     update(x_position: new_x, y_position: new_y)
     end
-  end
 
   def square_occupied?(new_x, new_y)
     piece = game.pieces.find_by(x_position: new_x, y_position: new_y)
```

Comment:
Nice one :) 

We could remove the if condition and have the second line be: `!piece.nil?`

Reply:
Just suggestions:
def space_occupied?(new_x, new_y)
    game.pieces.where(x_position: x, y_position: y).present?
end